src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx

```diff
@@ -1,18 +1,34 @@
 import React, { useState } from "react";
 import { useProfile } from "../../../utils";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
 
 export default function NewProfile() {
+  const navigate = useNavigate();
+  const { createProfile, profile, isLoading } = useProfile();
   const [firstName, setFirstName] = useState('');
   const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
-  const { createProfile } = useProfile()
 
   const handleCreateProfile = async () => {
     const newProfile = { firstName, lastName, email };
-    await createProfile(newProfile);
+    createProfile(newProfile);
   };
 
-  return (
+  useEffect(() => {
+    if (!isLoading) {
+      if (profile?.message == null && profile?.email != null) {
```

Comment:
I believe its not proper tracking, you should have something like `profile.isLoaded`

Reply:
kind of get your point. em.. to reducing confusing, I am going to use the `status` defined by useQuery:
```status: String
Will be:
loading if there's no cached data and no query attempt was finished yet.
error if the query attempt resulted in an error. The corresponding error property has the error received from the attempted fetch
success if the query has received a response with no errors and is ready to display its data. The corresponding data property on the query is the data received from the successful fetch or if the query's enabled property is set to false and has not been fetched yet data is the first initialData supplied to the query on initialization.```